pallets/corporate-actions/src/distribution/mod.rs

```diff
@@ -348,13 +347,17 @@ decl_module! {
         /// - `NotExpired` if `now < expiry`.
         #[weight = <T as Config>::DistWeightInfo::reclaim()]
         pub fn reclaim(origin, ca_id: CAId) {
-            // Ensure DID is the dist creator, they haven't reclaimed, and that expiry has passed.
-            let did = <Identity<T>>::ensure_perms(origin)?;
+            let PermissionedCallOriginData {
+                primary_did: caa,
+                secondary_key,
+                ..
+            } = <ExternalAgents<T>>::ensure_agent_asset_perms(origin.clone(), ca_id.ticker)?;
             let dist = Self::ensure_distribution_exists(ca_id)?;
-            ensure!(did == dist.from.did, Error::<T>::NotDistributionCreator);
-            let did = did.for_event();
+            ensure!(caa == dist.from.did, Error::<T>::NotDistributionCreator);
```

Comment:
Line should be removed and `NotDistCreator` is no longer needed as an error variant.

Reply:
made the change.